tsdb/chunks/head_chunks.go

```diff
@@ -632,7 +632,11 @@ func (cdm *ChunkDiskMapper) IterateAllChunks(f func(seriesRef, chunkRef uint64,
 				mmapFile.maxt = maxt
 			}
 
-			if err := f(seriesRef, chunkRef, mint, maxt, numSamples); err != nil {
+			chunkErrTpl := CorruptionErr{
+				Dir:       cdm.dir.Name(),
+				FileIndex: segID,
+			}
+			if err := f(seriesRef, chunkRef, mint, maxt, numSamples, chunkErrTpl); err != nil {
```

Comment:
We should be just wrapping this error with corruption error and return and nothing else. You don't need additional argument for `f`.

Reply:
Ops... sorry, my brainfart :p